asterixdb/asterix-algebra/src/main/java/org/apache/asterix/optimizer/rules/util/AsterixJoinUtils.java

```diff
@@ -116,7 +116,7 @@ public static void setJoinAlgorithmAndExchangeAlgo(AbstractBinaryJoinOperator op
                         spatialFunctionCallExists = true;
                     } else {
                         conditionExprs.add(exp);
-                        if (funcCallExp.getFunctionIdentifier().equals(BuiltinFunctions.ST_INTERSECTS)) {
+                        if (BuiltinFunctions.isSTFilterRefineFunction(funcCallExp.getFunctionIdentifier())) {
```

Comment:
Tin, isSTFilterRefineFunction() does not return true for ST_DISTANCE. I prepared this function to detect functions that are subject to FilterRefineSpatialJoin.java only.

Adding the ST_DISTANCE function check with an or statement can solve this.

Reply:
Thanks Akil. I fixed it.